plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java

```diff
@@ -269,6 +269,14 @@ private void appendSources(List<YamlSource> sources, String source) throws Confi
      */
     @Initializer(after = InitMilestone.EXTENSIONS_AUGMENTED, before = InitMilestone.JOB_LOADED)
     public static void init() throws Exception {
+        Long duration = Long.getLong(ConfigurationAsCode.class.getName() + ".initialDelay");
+        if (duration != null) {
+            try {
+                Thread.sleep(duration);
```

Comment:
Actually I think this sleep should be added later into the configure step.

To allow for room to introduce configuration of system properties.

Reply:
This seems like a better place to put the sleep in: https://github.com/jenkinsci/configuration-as-code-plugin/blob/cf4c8eb9781e26b4f8ebd954d6ac1a3607b7f048/plugin/src/main/java/io/jenkins/plugins/casc/ConfigurationAsCode.java#L719

Reply:
that is called by `configure` which is public and can be called when a config is updated to apply it, that would unnecessarily slow down reconfiguration, so would not be the correct place AFAICT (but I'm not a maintainer of the plugin so may be unaware of something else going on here)

Reply:
The problem with having the sleep inside init it does not allow for much flexibility if we wanted to introduce logic before the sleep happens.

The sleep would have to be modified so it only called during initialization.
A simple static volatile variable would do?

Reply:
Unless I am totally wrong @timja @oleg-nenashev ?

Reply:
> The problem with having the sleep inside init it does not allow for much flexibility if we wanted to introduce logic before the sleep happens.

The code does not support running anything before initialisation AFAICT so that would itself require a code change.  thus when you introduce that code you can move/change/refactor this sleep.

> The sleep would have to be modified so it only called during initialization.

this is the initialisation?
```java
    @Initializer(after = InitMilestone.EXTENSIONS_AUGMENTED, before = InitMilestone.JOB_LOADED)
    public static void init() throws Exception {      
```

Reply:
`init` is the initialisation, however my concern is having the sleep at the very beginning does not allow any room for introducing new functionality like #81 

Especially as shown in the example: https://github.com/jenkinsci/configuration-as-code-plugin/issues/81#issuecomment-526017173

Your of course welcome to push the problem till #81 is implemented.

Reply:
#81 needs to be run in a much earlier on lifecycle as plugins may use system properties to change their behaviour.  (It is also not clear how that would work with features in core that are controlled by properties as by the time of plugin loading it may be too late, but that is a different issue)